lib/dry/transaction/callable.rb

```diff
@@ -31,6 +31,9 @@ def initialize(operation)
       def call(*args, &block)
         if arity.zero?
           operation.(&block)
+        elsif args.last.is_a?(Hash)
```

Comment:
Hint: I'm not sure about how to solve this in a better way and could use some help. This line is to showcase why the deprecation warning is caused here. 

When `args` is an array with a last element being a hash, the splat operator pulls elements out of the array and the hash will be the last argument to the operation. If the operation is a method with keyword arguments, this will cause the deprecation warning `dry-transaction/lib/dry/transaction/callable.rb:35: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call`.

Any idea if this is missing some use cases, that also will lead to the same deprecation warning, but are not covered by the test suite and/or the `elsif`-block?

Any idea to improve the implementation here, I dislike the `elsif`-branch-style.

Reply:
Obviously this change does break older rubies. Looks like `**last` allows string hash keys to be splatted to keyword arguments in ruby  2.7 but not other rubies.

Reply:
I've added a a check to ruby27 to the condition. :)